### PR TITLE
Fix/multi vdc

### PR DESCRIPTION
**Pull request by andreaturli**
this PR contains the fixes for the #6 and support for multiple vCDs

Either we merge first #6 and I will then rebase that, or we discard #6 to merge just this one. Thoughts?

### CODE REVIEW

File: src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java [hunk 1]
```diff
@@ -438,4 +427,26 @@ public void suspendNode(String id) {
       api.getVmApi().suspend(id);
    }
 
+   private Vdc getVdc(String locationId) {
+      return api.getVdcApi().get(URI.create(locationId));
+   }
+
+   private Set<Vm> getAvailableVMsFromVAppTemplate(VAppTemplate vAppTemplate) {
+      return ImmutableSet.copyOf(filter(vAppTemplate.getChildren(), new Predicate<Vm>() {
+         // filter out vms in the vApp template with computer name that contains underscores, dots,
```
Comment (aledsage) [inline src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java:436]: Why do we do filter them out?
(I see the code was already there, just minor changes so this won't block merging).

File: src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapterLiveTest.java [hunk 2]
```diff
@@ -58,12 +58,17 @@ public void testCreateNodeWithGroupEncodedIntoNameThenStoreCredentials() {
       String name = "vm-" + new Random().nextInt();
 
       Template template = templateBuilder
-              .imageNameMatches("centos6.4x64") // TAI
+              .locationId("https://emea01.canopy-cloud.com/api/vdc/e931a09d-131f-4aaa-a667-efbe02eba428") // TAI2.0
```
Comment (aledsage) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapterLiveTest.java:61]: A shame to hard-code this within our tests. Means only those with a TAI account will be able to run the live tests, I presume?

File: src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapterLiveTest.java [hunk 3]
```diff
@@ -58,12 +58,17 @@ public void testCreateNodeWithGroupEncodedIntoNameThenStoreCredentials() {
...
+              // image
+              //.imageNameMatches("centos6.4x64") // TAI
               //.imageNameMatches("W2K12_STD_ENG_X64") // TAI
               //.imageNameMatches("cloudsoft-template")
+              .imageNameMatches("CentOS_66_x64_platform") // TAI2.0
               .build();
 
-      template.getOptions().networks("Deployment_Network_01"); // TAI
+      template.getOptions()
+              //.networks("Deployment_Network_01"); // TAI
```
Comment (aledsage) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapterLiveTest.java:70]: Are we only testing on TAI 2.0, or do we also have live tests against TAI 1.2?

File: src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceContextLiveTest.java [hunk 4]
```diff
@@ -64,19 +63,27 @@ public void testLaunchClusterWithMinDisk() throws RunNodesException {
...
+      try {
+         Set<? extends NodeMetadata> nodes = context.getComputeService().createNodesInGroup(name, 1, template);
+         node = Iterables.getOnlyElement(nodes);
          logger.debug("Created Node: %s", node);
          SshClient client = context.utils().sshForNode().apply(node);
          client.connect();
          ExecResponse hello = client.exec("mount");
          logger.debug(hello.getOutput().trim());
+      } finally {
          context.getComputeService().destroyNode(node.getId());
```
Comment (aledsage) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceContextLiveTest.java]: Guard  against `node == null`, or move the `nodes = ...` and `node = ...` outside of the try block.

**Comment by aledsage** [comment]
Looks good; only minor comments. Happy for this to be merged.

File: src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapterLiveTest.java (hunk 3 above)
Comment (andreaturli) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapterLiveTest.java:70]: I don't know if TAI is different from TAI1.2

File: src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapterLiveTest.java (hunk 2 above)
Comment (andreaturli) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapterLiveTest.java:61]: agreed, I should use probably `-Dtest.jclouds-vcloud-director.template` and remove those hard-coded values.

**Comment by andreaturli** [comment]
thanks @aledsage I'll merge as soon as travis builder is happy.

**Comment by aledsage** [comment]
Looks like travis is happy; merging.

**Pull request merged**